theano/compile/tests/test_builders.py

```diff
@@ -107,6 +107,19 @@ def test_shared_grad(self):
         fn = function([x, y, z], f)
         assert numpy.allclose(15.0 + s.get_value(),
                               fn(xv, yv, zv))
+    
+    def test_connection_pattern(self):
+        import numpy
+        x, y, z = T.matrices('xyz')
+        out1 = x * y
+        out2 = y * z
+
+        op = OpFromGraph([x ,y, z], [out1, out2], moe='FAST_RUN')
+        results = op.connection_pattern(None)
+        expect_result = [[True, False],
+                         [True, True],
+                         [False, True]]
+        assert results == expect_result
```

Comment:
I added another 2 test cases. Test is not passed.
Yet I don't know how theano deal with sharedvariable in connection_pattern().
If the ```expected_result``` in the last case right? If not, please provide the right one.